pkg/api/types.go

```diff
@@ -1603,6 +1607,10 @@ type Endpoints struct {
 
 	// The set of all endpoints is the union of all subsets.
 	Subsets []EndpointSubset
+
+	// Its value is the json representation of map[string(IP)][HostRecord]
+	// example: '{"10.245.1.6":{"HostName":"my-webserver"}}'
+	PodHostnames string `json:"podHostnames,omitempty"`
```

Comment:
This should be an actual field, not a string of a struct.  I think you can instead add an optional hostname field to EndpointAddress, next to IP.